# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerDebuggerChild.h
# Commit: 01fe98c08d10
# Full Hash: 01fe98c08d102c28e3a61209ea31834eff83202a
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-25 21:02:33
# Regressor Bug: 1899503
# File Overlap Count: 4
# Description:
#   Bug 1899503 - P3 PRemoteWorkerDebugger binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebugger binding.
#   
#   We are trying to keep the same logic debugger registration/unregistration to minimize the difference between the local debugger(the debugger on the content process main thread)
# ==============================================================================

diff -r 9bd2a82481f4 -r 01fe98c08d10 dom/workers/remoteworkers/RemoteWorkerDebuggerChild.h
--- a/dom/workers/remoteworkers/RemoteWorkerDebuggerChild.h	Tue Mar 25 09:13:20 2025 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerDebuggerChild.h	Tue Mar 25 09:13:20 2025 +0000
@@ -16,12 +16,16 @@
 
 class RemoteWorkerDebuggerChild final : public PRemoteWorkerDebuggerChild {
   friend class PRemoteWorkerDebuggerChild;
+  friend class WorkerPrivate;
 
  public:
   NS_INLINE_DECL_THREADSAFE_REFCOUNTING(RemoteWorkerDebuggerChild, final)
 
   explicit RemoteWorkerDebuggerChild(WorkerPrivate* aWorkerPrivate);
 
+  mozilla::ipc::IPCResult RecvRegisterDone();
+  mozilla::ipc::IPCResult RecvUnregisterDone();
+
   mozilla::ipc::IPCResult RecvInitialize(const nsString& aURL);
   mozilla::ipc::IPCResult RecvPostMessage(const nsString& aMessage);
   mozilla::ipc::IPCResult RecvSetDebuggerReady(const bool& aReady);
@@ -29,7 +33,7 @@
  private:
   ~RemoteWorkerDebuggerChild();
 
-  RefPtr<WorkerPrivate> mWorkerPrivate;
+  bool mIsInitialized{false};
 };
 
 }  // namespace mozilla::dom